src/libpriv/rpmostree-scripts.c

```diff
@@ -317,23 +344,261 @@ run_script (const KnownRpmScriptKind *rpmscript,
                               cancellable, error);
 }
 
+#ifdef BUILDOPT_HAVE_RPM_FILETRIGGERS
+static gboolean
+write_filename (FILE *f, GString *prefix,
+                GError **error)
+{
+  if (fwrite_unlocked (prefix->str, 1, prefix->len, f) != prefix->len)
+    return glnx_throw_errno_prefix (error, "fwrite");
+  if (fputc_unlocked ('\n', f) == EOF)
+    return glnx_throw_errno_prefix (error, "fputc");
+  return TRUE;
+}
+
+/* Used for %transfiletriggerin - basically an implementation of `find -type f` that
+ * writes the filenames to a tmpfile.
+ */
+static gboolean
+write_subdir (int dfd, const char *path,
+              GString *prefix,
+              FILE *f,
+              guint *inout_n_matched,
+              GCancellable *cancellable,
+              GError **error)
+{
+  /* This is an inlined copy of ot_dfd_iter_init_allow_noent()...if more users
+   * appear we should probably push to libglnx.
+   */
+  glnx_fd_close int target_dfd = glnx_opendirat_with_errno (dfd, path, TRUE);
+  if (target_dfd < 0)
+    {
+      if (errno != ENOENT)
+        return glnx_throw_errno_prefix (error, "opendirat");
+      /* Not early return */
+      return TRUE;
+    }
+  g_auto(GLnxDirFdIterator) dfd_iter = { 0, };
+  if (!glnx_dirfd_iterator_init_take_fd (&target_dfd, &dfd_iter, error))
+    return FALSE;
+
+  while (TRUE)
+    {
+      struct dirent *dent;
+
+      if (!glnx_dirfd_iterator_next_dent_ensure_dtype (&dfd_iter, &dent, cancellable, error))
+        return FALSE;
+      if (dent == NULL)
+        break;
+
+      const size_t origlen = prefix->len;
+      g_string_append_c (prefix, '/');
+      g_string_append (prefix, dent->d_name);
+      if (dent->d_type == DT_DIR)
+        {
+          if (!write_subdir (dfd_iter.fd, dent->d_name, prefix, f,
+                             inout_n_matched,
+                             cancellable, error))
+            return FALSE;
+        }
+      else
+        {
+          if (!write_filename (f, prefix, error))
+            return FALSE;
+          (*inout_n_matched)++;
+        }
+      g_string_truncate (prefix, origlen);
+    }
+
+  return TRUE;
+}
+
+/* Given a set of file trigger @patterns (really subdirectories), traverse the
```

Comment:
> (really subdirectories)

Is that how librpm treats them as well? Or do they do a simple prefix check on the full filename? The docs don't explicitly say they're directories.